src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1565,7 +1565,7 @@ private void WriteRedFactory(Node nd)
 
                         if (kinds.Count == 1)
                         {
-                            WriteLine($"if ({pname}.Kind() != SyntaxKind.{kinds[0].Name}) throw new ArgumentException(nameof({pname}));");
+                            WriteLine($"if (!{pname}.IsKind(SyntaxKind.{kinds[0].Name})) throw new ArgumentException(nameof({pname}));");
                         }
                         else
```

Comment:
@sharwell The code under `else` is using `Kind` in a switch, which can be replaced with `IsKind` that are ORed together, should I do that?

Reply:
That code seems fine.